cpp/src/backends/torch_scripted/handler/base_handler.cc

```diff
@@ -42,9 +42,24 @@ std::shared_ptr<torch::Device> BaseHandler::GetTorchDevice(
                                          load_model_request->gpu_id);
 }
 
+std::string BaseHandler::BuildRequestIdBatch(
```

Comment:
It is good to have a function for code isolation. However, we also need think about perf. A batch can be very big. it is not necessary to iterate the batch twice in preprrocessing. the batch id string can be built in one loop (ie. https://github.com/pytorch/serve/blob/1544bd3b5865e20f78fbb02f92a8af421354da79/cpp/src/backends/torch_scripted/handler/base_handler.cc#L73).

Reply:
Agreed, I initially considered this approach but realized that the `batch request id` may be used outside the context of the handler, for example in the `ModelInstance::Predict` method to record the `PredictionTime` metric: https://github.com/pytorch/serve/pull/1975/files#diff-5758cb72b743a174c7a467614854369a6daac8e24bd86de05f25173677ae15f4R94

```
    prediction_time_metric.AddOrUpdate(
        std::vector<std::string>{manifest_->GetModel().model_name, "Model"},
        handler_->BuildRequestIdBatch(request_batch), duration.count());
```

Therefore, created a helper method to build the batch request id in the base handler.

Any suggestions on how to better structure this?

Reply:
Updated implementation to iterate over `request_batch` and build `request_id_batch` only once in `base_handler`

Reply:
This can be solved in [predict function](https://github.com/pytorch/serve/blob/cpp_backend/cpp/src/backends/torch_scripted/torch_scripted_backend.cc#L78) too.

Reply:
Updated implementation to record both `HandlerTime` and `PredictionTime` metrics in `ModelInstance::Predict` method.